app/models/like.rb

```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+    belongs_to :author, class_name: 'User'
+    belongs_to :post
+  
+    after_create :increment_post_likes_counter
+  
+    private
+  
+    def increment_post_likes_counter
+      post.increment!(:likes_counter)
+    end
```

Comment:
- [ ]  Kindly I recommend you to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save `to call the #`increment_post_likes_counter `method every time a new like has been saved to database,

- `after_create` is a callback that is only triggered when a new object is created. It will not be triggered when an object is updated.

- `after_save`, on the other hand, is triggered every time an object is saved. This means it will be called both when an object is created and when it is updated.

- [x] In your case, if you use ` after_create :increment_post_likes_counter`, the `increment_post_likes_counter` method will only be called when a new like is created. If a like is updated, the counter will not change.

- [x] If you use` after_save :increment_post_likes_counter,` the `increment_post_likes_counter `method will be called every time a like is created or updated. This might be useful if the update can change whether the like is counted or not. So kindly I recommend you to use **` after_save :increment_post_likes_counter,`**
 ```
after_save : increment_post_likes_counter

  private

   def increment_post_likes_counter
      post.increment!(:likes_counter)
    end
```